### PR TITLE
docs: describe `push.branch-prefix`

### LINKED ISSUE #1958: Specify the naming or prefix for the `jj git push --change xxx`
Hi,

I've been using sapling and now trying to do the similar thing as push stack. I followed the docs and run commands like
```
jj git push --change XXX --change YYY
```

However, the created branch is with naming like
```
push-otznykxnuusx
push-vlxzuutruspk
```

My questions are:
1. Is it possible to specify the naming with the `--change` flag?
2. Or is it possible to specify a template, like `<username>/push-xxxxxx`, to make it easier to work in a collaborated repo?
3. Since we need to manually specify the base branch when creating the stack PRs with this method, is it possible to add some sort of ordinal naming like `push-42-otznykxnuusx` or `push-pr42-otznykxnuusx` so it's easier to spot out the ordering of multiple created branches / PRs?

Thanks!

### PR DESCRIPTION
**Pull request by martinvonz**
Closes #1958

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -448,6 +448,11 @@ branches. Since the local branch isn't created, the remote branch will be
 deleted if you push the branch with `jj git push --branch` or `jj git push
 --all`.
 
+### Prefix for generated branches on push
+
+`jj git push --change` generates branch names with a prefix of "push-" by
+default. You can pick a different prefix by setting `push.branch-prefix`.
```
Comment (ilyagr) [inline docs/config.md:454]: Add an example?

    push.branch-prefix="martinvonz/push-"

Comment (ilyagr) [inline docs/config.md:454]: Not for this PR: the inconsistency in naming of `git.auto-local-branch` and `push.branch-prefix` is mildly annoying.

Perhaps they could be `branch.push-prefix` and `branch.auto-create-local`, or something along these lines?

**Review by ilyagr — APPROVED** [review]

File: docs/config.md (hunk 1 above)
Comment (ilyagr) [reply docs/config.md:454]: Our you could do `your_initials/push-`.

Comment (martinvonz) [reply docs/config.md]: Good idea. Done.

Comment (martinvonz) [reply docs/config.md]: Good point. It probably belongs under `[git]`.

**Pull request merged**